lru-disk-cache/src/lib.rs

```diff
@@ -202,6 +202,23 @@ impl LruDiskCache {
             })
     }
 
+    /// Add a file by calling `with` with the open `File` corresponding to the cache at path `key`.
+    pub fn insert_with<K: AsRef<OsStr>, F: FnOnce(File) -> io::Result<()>>(&mut self, key: K, with: F) -> Result<()> {
```

Comment:
I'm not quite sure on the precise differences of this vs. `insert_by`. Can this just call `insert_by`, passing a closure that opens the file and calls `with(f)`?